fcrepo/fcrepo
# Adds support for side-loading of new resources.

**Adds support for side-loading of new resources**
* * *

**JIRA Ticket**: Resolves: https://jira.lyrasis.org/browse/FCREPO-3332

* Other Relevant Links (Mailing list discussion, related pull requests, etc.)
Two follow on tickets for reindexing existing resources and ensuring that newly created, updated, or deleted resources emit events when they are reindexed.
* https://jira.lyrasis.org/browse/FCREPO-3553
* https://jira.lyrasis.org/browse/FCREPO-3554

# What does this Pull Request do?
This PR adds the /fcr:reindex endpoint on all resources. So for  example http://localhost:8080/my-resource/fcr:reindex is now exposed.
Eventually this endpoint will service on demand reindexing. With this PR,  this endpoint is used to accomplish "side-loading" which is the addition of new resources to your Fedora repository by adding OCFL objects directly to the OCFL layout.

# How should this be tested?
There is an integration test covering this feature.  However if you want try it manually, you can start up a fresh fedora and then copy the entire contents of `fcrepo-http-api/src/test/resources/reindex-test` into your ocfl-root directory (fcrepo-home/data/ocfl-root).

The id of the new ocfl resource is "info:fedora/container1" (as you'll notice in the inventory.json file).

To side load the resource, simply do the following:
```
curl -v -u fedoraAdmin:fedoraAdmin -X POST http://localhost:8080/rest/container1/fcr:reindex
```
It should return a 204. 

Then do
```
curl -v -u fedoraAdmin:fedoraAdmin http://localhost:8080/rest/container1
```
to verify that it was successfully side loaded.


# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo/committers
